From: Jhonny Oliveira <jhonny.oliveira@gmail.com>
Date: Sat, 15 Mar 2025 09:03:11 +0100
Subject: Disable wxWidgets uncaught exception handling

Bug: https://freefilesync.org/forum/viewtopic.php?t=10794
Forwarded: not-needed
Last-Update: 2023-10-29

wxWidgets' exception handling gets in the way of debugging by
handling uncaught exceptions, thous hiding their point of origin.
This causes the build to fail:
application.cpp:249:2: error: #error why is wxWidgets uncaught exception handling enabled!?
 249 | #error why is wxWidgets uncaught exception handling enabled!?
     |  ^~~~~
compilation terminated due to -Wfatal-errors.
Last-Update: 2023-10-29
---
 FreeFileSync/Source/RealTimeSync/application.cpp | 3 ---
 FreeFileSync/Source/application.cpp              | 3 ---
 2 files changed, 6 deletions(-)

diff --git a/FreeFileSync/Source/RealTimeSync/application.cpp b/FreeFileSync/Source/RealTimeSync/application.cpp
index 26fb48f..ae3aea8 100644
--- a/FreeFileSync/Source/RealTimeSync/application.cpp
+++ b/FreeFileSync/Source/RealTimeSync/application.cpp
@@ -201,9 +201,6 @@ wxLayoutDirection Application::GetLayoutDirection() const { return languageLayou
 
 int Application::OnRun()
 {
-#if wxUSE_EXCEPTIONS
-#error why is wxWidgets uncaught exception handling enabled!?
-#endif
 
     //exception => Windows: let it crash and create mini dump!!! Linux/macOS: std::exception::what() logged to console
         [[maybe_unused]] const int rc = wxApp::OnRun();
diff --git a/FreeFileSync/Source/application.cpp b/FreeFileSync/Source/application.cpp
index 5b2d0cb..15eb725 100644
--- a/FreeFileSync/Source/application.cpp
+++ b/FreeFileSync/Source/application.cpp
@@ -220,9 +220,6 @@ wxLayoutDirection Application::GetLayoutDirection() const { return languageLayou
 
 int Application::OnRun()
 {
-#if wxUSE_EXCEPTIONS
-#error why is wxWidgets uncaught exception handling enabled!?
-#endif
 
     //exception => Windows: let it crash and create mini dump!!! Linux/macOS: std::exception::what() logged to console
         [[maybe_unused]] const int rc = wxApp::OnRun();
